[Koh Su En] iP

seksek13 opened the pull request:
# Update Duke
> “Your mind is for having ideas, not holding them.” – David Allen [source](https://dansilvestre.com/productivity-quotes)

**_Level Completed_** 👍 :
1. Level 1
    - Greet, Echo, Exit 
2. Level 2
    - Add, List
3. Level 3
   - Mark as Done
4. Level 4
    - ToDos, Events, Deadlines
5. Level 5
    - Handle Errors
6. Level 6
     - Delete
7. Level 7
    - Save
8. Level 8
    - Dates and Times
9. Level 9
    - Find
10. ~~Level 10~~ 

If you Java programmer, you can use it to practice Java too. Here's the main method:
```js
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

`Array.prototype.map()`

**Features:**
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

seksek13 committed:
Level-1

seksek13 committed:
Level-2

seksek13 committed:
mark as done

seksek13 committed:
Level-3

seksek13 committed:
Level-4

seksek13 committed:
Level-4

seksek13 committed:
A-TextUiTesting

seksek13 committed:
Level-5

seksek13 committed:
Level-6

seksek13 committed:
Change system name

seksek13 committed:
Level-7

seksek13 committed:
Level-8

seksek13 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

seksek13 committed:
A-MoreOOP

seksek13 committed:
A-Packages

seksek13 committed:
Add time to date

seksek13 committed:
A-JUnit

seksek13 committed:
Correct Unmark function

seksek13 committed:
A-JavaDoc

seksek13 committed:
A-CodingStandard

seksek13 committed:
Level-9

seksek13 committed:
Merge tag 'A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Deadline.java
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Task.java
#	src/main/java/duke/Todo.java

seksek13 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java

seksek13 commented:
# Update Duke
> “Your mind is for having ideas, not holding them.” – David Allen [source](https://dansilvestre.com/productivity-quotes)

**_Level Completed_** 👍 :
1. Level 1
    - Greet, Echo, Exit 
2. Level 2
    - Add, List
3. Level 3
   - Mark as Done
4. Level 4
    - ToDos, Events, Deadlines
5. Level 5
    - Handle Errors
6. Level 6
     - Delete
7. Level 7
    - Save
8. Level 8
    - Dates and Times
9. Level 9
    - Find
10. ~~Level 10~~ 

If you Java programmer, you can use it to practice Java too. Here's the main method:
```js
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

`Array.prototype.map()`

**Features:**
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

seksek13 committed:
Merge branch 'add-gradle-support' of https://github.com/seksek13/ip into add-gradle-support

seksek13 committed:
A-CheckStyle

aaron-ljx commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+    /**
+     * Deadline task constructor
+     *
+     * @param description description of command
```
it's a little hard to read without any separators to separate the parameter name and your comment on what it means. perhaps you can consider @param description: description of command for example.

aaron-ljx commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+    /**
+     * Parse date into MMM dd yyyy format
+     *
+     * @return date in MMM dd yyyy hh:mm a format
```
I think it's okay not to repeat yourself here since you already said that the date is parsed into MM dd yyyy format.

ovidharshini replied:
@return's description can probably be changed to something simpler so as to avoid repetition

aaron-ljx commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,125 @@
+    /**
+     * Method to run entire programme
+     */
+    public void run() {
```
this method is very long. perhaps you could make it shorter by putting some of the logic in another method instead.

aaron-ljx commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,94 @@
+     * @return String list to store required info
+     * @throws DukeException
+     */
+    public static String[] parse(String command) throws DukeException {
```
perhaps you could consider using a switch statement instead of else-ifs? since it is easier to read through a switch statement and debug any possible errors compared to else-ifs.

aaron-ljx reviewed (commented):
Overall there is good use of OOP in the code. However, there are some minor issues in terms of the readability of the code which can be easily fixed with some light refactoring.

seksek13 committed:
Level 10 Add GUI

ovidharshini commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+package duke;
+import java.time.LocalDateTime;
```
Consider adding a line break between the packaging and import statements, perhaps

ovidharshini commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,113 @@
+        }
+    }
+
+    /**
+     * Method to run entire programme
+     */
```
Do include @param, @return, and @throw tags

ovidharshini commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,113 @@
+        try {
+            String[] commands = Parser.parse(input);
+            if (commands[0].equals("bye")) {
+                return ui.showBye();
+
+            } else if (commands[0].equals("list")) {
```
Spacing could be more consistent in-line with the rest of the if-else statements here. You might want to consider using switch-case here to better differentiate levels of nesting as well.

ovidharshini commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,113 @@
+                    count++;
+                }
+                return ui.showFindTask(t);
+            }else{
```
Perhaps add a whitespace character on either sides of the 'else' keyword here to comply with Java coding standards?

ovidharshini commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,7 @@
+package duke;
+public class DukeException extends Exception {
```
Consider leaving a line break after packaging for consistency

ovidharshini commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,118 @@
+     *Overrides toString method
+     *
+     * @return tasks in task list
+     */
+    public String toString() {
```
You might want to add an override tag here?

ovidharshini commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,123 @@
+    /**
+     * Read user input
+     *
+     * @return
```
Do specify the returned element here

ovidharshini commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,123 @@
+
+    private Scanner input = new Scanner(System.in);
+
+    /**
+     * Show welcome message
+     */
+    public String showWelcomeMessage(){
```
Consider adding @return String for this function since the return type is not void. I noticed this in other places as well.

ovidharshini reviewed (commented):
LGTM, just a few nits here and there! The coding style was overall in line with Java conventions. Do have more consistent Javadoc documentation and you should be good to merge

seksek13 committed:
Reformat code to match checkstyle and edit dialog style

seksek13 committed:
Update code to match checkstyle

seksek13 committed:
There are some unchecked assumption used in the code

These assumptions needs to be checked to minimise bugs

By using assertions, we can check whether these assumptions are true

If it fails, it will return an error message

seksek13 committed:
A-CodeQualitty

seksek13 committed:
Edit DateTime format accepted in Parser

There is an error in the previous update - A-CodeQuality, which leads to system accepting wrong format of date and time keyed in by user.

This change helps to make the system more consistent to the previous versions and prevent confusion

seksek13 committed:
Merge pull request #1 from seksek13/branch-A-Assertions

There are some unchecked assumption used in the code

seksek13 committed:
Merge pull request #2 from seksek13/branch-A-CodeQuality

Branch a code quality

seksek13 committed:
Add Reminder Function

Date and time entered by the user for deadline and event tasks are not in use properly.

Create reminder functions that allows user to key in dates and find tasks that will be due by that date.

Helps to make a better use of the date and time entered by the user, making more sense of the user input

seksek13 committed:
Update README.md

seksek13 committed:
Update README.md

seksek13 committed:
Minor changes to code and added Ui.png

seksek13 committed:
Merge branch 'master' of https://github.com/seksek13/ip

* 'master' of https://github.com/seksek13/ip:
  Update README.md
  Update README.md

seksek13 committed:
Set theme jekyll-theme-tactile

seksek13 committed:
Set theme jekyll-theme-merlot

seksek13 committed:
Changes some gui to look more natural

seksek13 committed:
Merge branch 'master' of https://github.com/seksek13/ip

* 'master' of https://github.com/seksek13/ip:
  Set theme jekyll-theme-merlot
  Set theme jekyll-theme-tactile

seksek13 committed:
Update README.md
